keyserver/src/main/scala/com/advancedtelematic/tuf/keyserver/roles/RootRoleKeyEdit.scala

```diff
@@ -2,61 +2,46 @@ package com.advancedtelematic.tuf.keyserver.roles
 
 import akka.http.scaladsl.util.FastFuture
 import com.advancedtelematic.libtuf.data.TufDataType.{KeyId, RepoId, TufKeyPair}
-import com.advancedtelematic.tuf.keyserver.db.{KeyRepositorySupport}
+import com.advancedtelematic.tuf.keyserver.db.{KeyRepository, KeyRepositorySupport}
 import com.advancedtelematic.libtuf.data.TufCodecs._
 import com.advancedtelematic.libtuf.data.ClientCodecs._
 
 import scala.async.Async.{async, await}
 import scala.concurrent.{ExecutionContext, Future}
 import com.advancedtelematic.tuf.keyserver.db.KeyRepository.KeyNotFound
 import com.advancedtelematic.libtuf.data.TufDataType.RoleType.RoleType
-import com.advancedtelematic.tuf.keyserver.vault.VaultClient
-import com.advancedtelematic.tuf.keyserver.vault.VaultClient.VaultResourceNotFound
 import slick.jdbc.MySQLProfile.api._
-import com.advancedtelematic.libtuf.data.RootManipulationOps._
 import cats.implicits._
 import com.advancedtelematic.libtuf.data.RootManipulationOps._
-import com.advancedtelematic.tuf.keyserver.http.Errors
 
-class RootRoleKeyEdit(vaultClient: VaultClient)
+class RootRoleKeyEdit()
                       (implicit val db: Database, val ec: ExecutionContext)
   extends KeyRepositorySupport {
-  val roleSigning = new RoleSigning(vaultClient)
-  val signedRootRole = new SignedRootRoles(vaultClient)
+  val roleSigning = new RoleSigning()
+  val signedRootRole = new SignedRootRoles()
 
   def deletePrivateKey(repoId: RepoId, keyId: KeyId): Future[Unit] = for {
     _ <- ensureIsRepoKey(repoId, keyId)
-    _ <- vaultClient.deleteKey(keyId)
+    _ <- keyRepo.delete(keyId)
   } yield ()
 
   def findAllKeyPairs(repoId: RepoId, roleType: RoleType): Future[Seq[TufKeyPair]] = async {
```

Comment:
I find this method harder to read then the one below, although they are pretty similar is there a way to simplify it?

Reply:
Simplest I could do it:

```scala
  def findAllKeyPairs(repoId: RepoId, roleType: RoleType): Future[Seq[TufKeyPair]] =
    for {
      rootRole <- signedRootRole.find(repoId)
      targetKeyIds = rootRole.signed.roleKeys(roleType).map(_.id)
      keyPairsT <- keyRepo.findAll(targetKeyIds).map(_.map(_.toTufKeyPair))
      keyPairs <- Future.fromTry(keyPairsT.toList.sequence)
    } yield keyPairs
```

would that be better?

Reply:
or

```scala
  def findAllKeyPairs(repoId: RepoId, roleType: RoleType): Future[Seq[TufKeyPair]] =
    for {
      rootRole <- signedRootRole.find(repoId)
      targetKeyIds = rootRole.signed.roleKeys(roleType).map(_.id)
      dbKeys <- keyRepo.findAll(targetKeyIds)
      keyPairsT = dbKeys.map(_.toTufKeyPair)
      keyPairs <- Future.fromTry(keyPairsT.toList.sequence)
    } yield keyPairs
```

to avoid the double `map`

Reply:
Yeah, the last one looks clearer to me.